### PR TITLE
Use tweaked ECDSA blinding

### PR DESCRIPTION
**Pull request by chris-wood**
See the [corresponding PR](https://github.com/cloudflare/pat-go/pull/6).

cc @tlepoint, @FredericJacobs

### CODE REVIEW

File: draft-dew-cfrg-signature-key-blinding.md [hunk 1]
```diff
@@ -255,26 +255,30 @@ BlindKeySign(skS, skB, msg) works as follows:
 
 This section describes implementations of BlindPublicKey, UnblindPublicKey, and BlindKeySign as 
 functions implemented on top of an existing {{ECDSA}} implementation. In the descriptions below,
-let L be the order of the corresponding elliptic curve group used for ECDSA. For example, for 
-P-256, L = 2^256 - 2^224 + 2^192 + 2^96 - 1.
+let p be the order of the corresponding elliptic curve group used for ECDSA. For example, for 
+P-256, p = 2^256 - 2^224 + 2^192 + 2^96 - 1.
```
Comment (tlepoint) [inline draft-dew-cfrg-signature-key-blinding.md:259]: This value is the prime modulus `p` of the curve, not its order `n`. (Cf. Section 4.2.1.3 https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-186-draft.pdf)

File: draft-dew-cfrg-signature-key-blinding.md [hunk 2]
```diff
@@ -255,26 +255,30 @@ BlindKeySign(skS, skB, msg) works as follows:
...
+BlindPublicKey(pk, skB)   = ScalarMult(pk, HashToScalar(skB))
+UnblindPublicKey(pk, skB) = ScalarMult(pk, ModInverse(HashToScalar(skB), p))
 ~~~
 
 ## BlindKeySign
 
 BlindKeySign transforms the signing key skS by the private key skB into a new
 signing key, skR, and then invokes the existing ECDSA signing procedure. More
-specifically, skR = skS \* skR (mod L).
+specifically, skR = skS \* HashToScalar(skR) (mod L).
```
Comment (tlepoint) [inline draft-dew-cfrg-signature-key-blinding.md:281]:
```suggestion
specifically, skR = skS \* HashToScalar(skB) (mod p).
```

**Review by tlepoint — CHANGES_REQUESTED** [review]

File: draft-dew-cfrg-signature-key-blinding.md (hunk 1 above)
Comment (chris-wood) [reply draft-dew-cfrg-signature-key-blinding.md:259]: Oops, existing bug. Thanks for flagging! (Also, do you have these memorized? 😆 )

Comment (chris-wood) [inline draft-dew-cfrg-signature-key-blinding.md:259]:
```suggestion
P-256, p = 115792089210356248762697446949407573529996955224135760342422259061068512044369.
```

**Review by tedeaton — APPROVED** [review]

**Comment by chris-wood** [comment]
@jedisct1 please let me know what you think!

**Review by tlepoint — APPROVED** [review]

**Pull request merged**